crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1220,9 +1279,26 @@ pub struct ClassType<'db> {
     definition: Definition<'db>,
 
     body_scope: ScopeId<'db>,
+
+    is_builtin: Option<BuiltinType>,
```

Comment:
As mentioned above, I'd name this just `builtin` -- `is_` prefix suggests a boolean.